packages/terra-application/src/application-base/ApplicationBase.jsx

```diff
@@ -122,12 +122,17 @@ const ApplicationBase = ({
 
   const { localeOverride, themeOverride } = useTestOverrides(); // Allows us to test deployed applications in different locales.
 
+  let density = themeDensity || themeConfig?.density;
+  if (!density && themeName === 'redwood-theme') {
```

Comment:
Small suggestion: If we swap the conditions
```suggestion
  if (themeName === 'redwood-theme' && !density) {
```
then it will only check density when the theme is redwood since it will short-circuit.